x509util/extensions.go

```diff
@@ -389,6 +412,22 @@ func (s SubjectAlternativeName) RawValue() (asn1.RawValue, error) {
 			IsCompound: true,
 			Bytes:      rdn,
 		}, nil
+	case UPNType:
+		if s.Value == "" {
+			return zero, errors.New("error parsing UserPrincipalName SAN: empty Value is not allowed")
+		}
+		rawBytes, err := marshalExplicitValue(s.Value, "utf8")
+		if err != nil {
+			return zero, errors.Wrapf(err, "error marshaling ASN1 value %q", s.Value)
+		}
+		upnBytes, err := asn1.MarshalWithParams(otherName{
+			TypeID: oidUserPrincipalName,
+			Value:  asn1.RawValue{FullBytes: rawBytes},
+		}, "tag:0")
+		if err != nil {
+			return zero, errors.Wrap(err, "unable to Marshal UserPrincipalName SAN")
```

Comment:
In line with other (un)marshaling errors:
```suggestion
			return zero, errors.Wrap(err, "error marshaling UserPrincipalName SAN")
```

Reply:
Fixed with b724aa2